src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,379 @@
+                        AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_PACKAGE_ALREADY_EXISTS);
+                    }
+                    else
+                    {
+                        AICLI_LOG(CLI, Info, << "Overriding registry match check...");
+                        context.Reporter.Warn() << Resource::String::PortableRegistryCollisionOverridden << std::endl;
+                    }
+                }
+            }
+
+            AICLI_LOG(CLI, Info, << "Begin writing to Uninstall registry.");
+            uninstallEntry.SetValue(PortableValueName::DisplayName, entry.DisplayName);
+            uninstallEntry.SetValue(PortableValueName::DisplayVersion, entry.DisplayVersion);
+            uninstallEntry.SetValue(PortableValueName::Publisher, entry.Publisher);
+            uninstallEntry.SetValue(PortableValueName::InstallDate, Utility::GetCurrentDateForARP());
+            uninstallEntry.SetValue(PortableValueName::URLInfoAbout, manifest.DefaultLocalization.Get<Manifest::Localization::PackageUrl>());
+            uninstallEntry.SetValue(PortableValueName::HelpLink, manifest.DefaultLocalization.Get<Manifest::Localization::PublisherSupportUrl>());
+            uninstallEntry.SetValue(PortableValueName::UninstallString, L"winget uninstall --product-code " + productCode);
+            uninstallEntry.SetValue(PortableValueName::WinGetInstallerType, ConvertToUTF16(InstallerTypeToString(InstallerTypeEnum::Portable)));
+            uninstallEntry.SetValue(PortableValueName::WinGetPackageIdentifier, manifest.Id);
+            uninstallEntry.SetValue(PortableValueName::WinGetSourceIdentifier, sourceIdentifier);
+            uninstallEntry.SetValue(PortableValueName::PortableTargetFullPath, GetPortableTargetFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, GetPortableSymlinkFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::SHA256, Utility::SHA256::ConvertToWideString(context.Get<Execution::Data::HashPair>().second));
+            uninstallEntry.SetValue(PortableValueName::InstallLocation, GetPortableTargetDirectory(context).wstring());
+            AICLI_LOG(CLI, Info, << "Writing to Uninstall registry complete.");
+        }
+
+        void MovePortableExeAndCreateSymlink(Execution::Context& context)
+        {
+            const std::filesystem::path& installerPath = context.Get<Execution::Data::InstallerPath>();
+            const std::filesystem::path& targetFullPath = GetPortableTargetFullPath(context);
+            const std::filesystem::path& symlinkFullPath = GetPortableSymlinkFullPath(context);
+            const std::filesystem::path& targetDirectory = GetPortableTargetDirectory(context);
+
+            bool isDirectoryCreated = false;
+            if (!std::filesystem::exists(targetDirectory) || !std::filesystem::is_directory(targetDirectory))
```

Comment:
If `std::filesystem::is_directory` is false, we aren't going to be able to make it a directory.  That is an error condition that you could have left in the other location.

Reply:
I removed the checks and just call `createdirectories` since any errors will throw an exception which we handle.